app/src/main/java/com/example/mystore/ui/home/HomeScreen.kt

```diff
@@ -55,26 +57,26 @@ fun HomeScreen(destinationsNavigator: DestinationsNavigator) {
                 .padding(horizontal = dimensionResource(id = R.dimen.padding_medium_horizontal))
         ) {
             Search(
-                input = viewModel.input,
+                input = viewModel.input.collectAsState().value,
                 onInputChanged = { viewModel.filterProducts(it) },
                 onInputCleared = { viewModel.clearInput() }
             )
 
             Filter(
                 filtersCount = viewModel.filtersCount,
                 currentFilters = viewModel.filtersModel,
-                categories = viewModel.categories
+                categories = viewModel.categories.collectAsState().value
```

Comment:
It might be easier for you to collect the state up at the top of the file first. You can also use the `by` syntax to expose the value directly.

`val categories by viewModel.categories.collectAsState()`

And here you would just have a `categories = categories`.